sdk/servicebus/service-bus/src/session/messageSession.ts

```diff
@@ -838,11 +834,11 @@ export class MessageSession extends LinkEntity<Receiver> {
     context: ConnectionContext,
     entityPath: string,
     sessionId: string | undefined,
-    options?: MessageSessionOptions
+    options: MessageSessionOptions
   ): Promise<MessageSession> {
     throwErrorIfConnectionClosed(context);
     const messageSession = new MessageSession(context, entityPath, sessionId, options);
-    await messageSession._init(options?.abortSignal);
+    await messageSession._init(options);
```

Comment:
I'm starting have a hard time keeping track of where the client options on the ServiceBusClient get trickled down to. If the user specifies requestOptions on the ServiceBusClient, will they make their way down to the `messageSession._init` call?

Reply:
Yes, they do.
We call `messageSession._init` inside the `MessageSession.create()` method.
`MessageSession.create()` is invoked only at `SBClient.acceptSession` and `SBClient.acceptNextSession`, which is where I plug the clientOptions. 🙂

I agree it is tough to track all these.

If it makes it any easier.. I have managed to cover all the paths available to reach `getToken` from SBClient with the following test-suite. https://github.com/Azure/azure-sdk-for-js/blob/998f0828a197808145d6a9bba71464d0839a074a/sdk/servicebus/service-bus/test/internal/node/operationOptionsForAMQP.spec.ts#L479-L480
Please check that at let me know if I missed anything.

Reply:
To @chradek's point, I'm having the same difficulty. I wish it was just something that implicitly happened instead of something we need to instrument at each call site but testing will at least prove it is happening (for now).

Reply:
I had to lay down the whole tree of paths on a paper to understand where to pass the options.
I don't think we can do any better given the huge codebase and the much convoluted paths.